export/export_pipeline/compute_collision_representation.py

```diff
@@ -0,0 +1,427 @@
+            acd_args['decompose_method'] = 'none'
+            print(f'{obj_name} is small, volume = {volume:.5f}, will not decompose')
+
+        # save convex hull
+        hull_mesh: trimesh.Trimesh = trimesh.convex.convex_hull(mesh)
+        hull_filename = osp.join(mesh_dir, 'convex_hull.stl')
+        hull_mesh.export(hull_filename)
+        print(f'Convex hull {hull_filename} saved.')
+        
+        if acd_args['decompose_method'] == 'coacd':
+            # Call CoACD.
+            if coacd_args['auto_threshold']:
+                model = coacd_args['threshold_model']
+                threshold = max(0.02, volume*model[0] + model[1])
+            else:
+                threshold = coacd_args['threshold']
+            mesh = coacd.Mesh(mesh.vertices, mesh.faces)
+            result = coacd.run_coacd(
+                mesh,
+                threshold=threshold,
+                max_convex_hull=coacd_args['max_convex_hull'],
+                preprocess_mode=coacd_args['preprocess_mode'],
+                preprocess_resolution=coacd_args['preprocess_resolution'],
+                resolution=coacd_args['resolution'],
+                mcts_nodes=coacd_args['mcts_nodes'],
+                mcts_iterations=coacd_args['mcts_iterations'],
+                mcts_max_depth=coacd_args['mcts_max_depth'],
+                pca=coacd_args['pca'],
+                merge=coacd_args['merge'],
+                seed=coacd_args['seed'],
+                )
+
+            mesh_parts = []
+            for vs, fs in result:
+                # remove faces with duplicate vertices
+                fs = [f for f in fs if len(set(f)) == 3]
+                # remove duplicate faces
+                fs_set = set()
+                final_fs = []
+                for f in fs:
+                    ff = frozenset(f)  # frozenset() also sorts
+                    if ff in fs_set:
+                        fs_set.remove(ff)
+                    else:
+                        fs_set.add(ff)
+                        final_fs.append(f)
+                mesh = trimesh.Trimesh(vs, final_fs)
+                if mesh.is_volume:
+                    mesh_parts.append(mesh)
+
+            object_assembled = trimesh.Scene()
+            np.random.seed(0)
+            i = 0
+            for p in mesh_parts:
+                p.visual.vertex_colors[:, :3] = (np.random.rand(3) * 255).astype(np.uint8)
+                object_part = trimesh.Scene()
+                object_part.add_geometry(p)
+                filename = osp.join(decompose_dir, f"{i:03d}{params['common']['CONVEX_DECOMPOSITION_EXT']}")
+                object_part.export(filename)
+                object_assembled.add_geometry(p)
+                i+=1
+            if not object_assembled.is_empty:
+                object_assembled.export(osp.join(decompose_dir, 'assembled.obj'))
+                decomposed = True
+        elif acd_args['decompose_method'] == 'vhacd':
+            if not vhacd_args['enable']:
```

Comment:
Alternatively, if we were willing to remove V-HACD entirely, we could simplify this file significantly. My current thinking is that it is a good idea to leave V-HACD enabled for now as a debug/diagnostic tool, because it is the default algorithm used in Unreal, so it is arguably a useful baseline.